### PR TITLE
bpo-31107: Fix copyreg mangled slot names calculation.

### PR DESCRIPTION
**Pull request by ShaneHarvey**
https://bugs.python.org/issue31107

https://bugs.python.org/issue31107

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately we couldn't find an account corresponding to your GitHub username on [bugs.python.org](http://bugs.python.org/) (b.p.o) to verify you have signed the CLA (this might be simply due to a missing "GitHub Name" entry in your b.p.o account settings). This is necessary for legal reasons before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

File: Lib/copyreg.py [hunk 1]
```diff
@@ -128,7 +128,7 @@ class found there.  (This assumes classes don't modify their
                         continue
                     # mangled names
                     elif name.startswith('__') and not name.endswith('__'):
-                        names.append('_%s%s' % (c.__name__, name))
+                        names.append('_%s%s' % (c.__name__.lstrip('_'), name))
```
Comment (serhiy-storchaka) [inline Lib/copyreg.py]: The name shouldn't be mangled if `c.__name__.lstrip('_')` is an empty string.

**Comment by serhiy-storchaka** [comment]
Please add a [NEWS entry](https://docs.python.org/devguide/committing.html#what-s-new-and-news-entries).

File: Lib/copyreg.py (hunk 1 above)
Comment (ShaneHarvey) [reply Lib/copyreg.py]: Nice catch, I wasn't aware of that case.

**Review by ShaneHarvey — COMMENTED** [review]
Added a NEWS entry and a test with an underscore only class name.

**Review by serhiy-storchaka — APPROVED** [review]
LGTM.

Please sign the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) and add your name into `Misc/ACKS`.

**Comment by ShaneHarvey** [comment]
Thanks for the review @serhiy-storchaka. I signed the PSF contributor agreement yesterday and my GitHub is linked to my bugs.python.org account. Does it take some time for @the-knights-who-say-ni to notice?

**Comment by serhiy-storchaka** [comment]
Usually it takes a one working day (as in this case).

**Comment by serhiy-storchaka** [comment]
Thank you for your contribution @ShaneHarvey! Do you mind to [backport](https://docs.python.org/devguide/committing.html#backporting-changes-to-an-older-version) these changes to 3.6 and 2.7?

**Comment by ShaneHarvey** [comment]
Sure I can backport it to those branches.

**Pull request merged**